python_modules/dagster/dagster/_core/external_execution/context.py

```diff
@@ -54,8 +55,8 @@ def build_external_execution_context(
             _convert_time_window(partition_time_window) if partition_time_window else None
         ),
         run_id=context.run_id,
-        job_name=context.job_def.name,
```

Comment:
how is a `BoundOpExecutionContext` getting here in the first place? Shouldn't it always be bound by the time it gets to user space?

Reply:
Confused by your question, it sounds like the first question should say "how is an `UnboundOpExecutionContext`...

`BoundOpExecutionContext` is the name of the passed class as context when directly invoking, it has nothing to do with the externals work.

Reply:
Yeah I got confused and meant `UnboundOpExecutionContext` because it is confusing that we have both `BoundOpExecutionContext` and `OpExecutionContext` as non-abstract classes